app/controllers/reactions_controller.rb

```diff
@@ -4,37 +4,49 @@ class ReactionsController < ApplicationController
 
   def index
     skip_authorization
+
     if params[:article_id]
       id = params[:article_id]
-      reactions = if session_current_user_id.present?
-                    Reaction.where(reactable_id: id,
-                                   reactable_type: "Article",
-                                   user_id: session_current_user_id).
-                      where("points > ?", 0)
+
+      reactions = if session_current_user_id
+                    Reaction.positive.
+                      where(
+                        reactable_id: id,
+                        reactable_type: "Article",
+                        user_id: session_current_user_id,
+                      )
                   else
-                    []
+                    Reaction.none
                   end
-      render json:
-      {
-        current_user: { id: session_current_user_id },
-        article_reaction_counts: Reaction.count_for_article(id),
-        reactions: reactions
-      }.to_json
+
+      result = { article_reaction_counts: Reaction.count_for_article(id) }
     else
-      comments = Comment.where(
-        commentable_id: params[:commentable_id],
-        commentable_type: params[:commentable_type],
-      ).select(%i[id positive_reactions_count])
-      comment_ids = comments.map(&:id)
-      reaction_counts = comments.map { |c| { id: c.id, count: c.positive_reactions_count } }
-      reactions = session_current_user_id ? cached_user_positive_reactions(current_user).where(reactable_id: comment_ids) : []
-      render json:
-        {
-          current_user: { id: session_current_user_id },
-          positive_reaction_counts: reaction_counts,
-          reactions: reactions
-        }.to_json
+      comments = Comment.
+        where(commentable_id: params[:commentable_id], commentable_type: params[:commentable_type]).
+        select(%i[id positive_reactions_count])
+
+      # iterating only once
```

Comment:
here we were iterating on comments objects twice even though we could iterate once :D